packages/ckeditor5-table/src/tableclipboard.js

```diff
@@ -74,4 +81,220 @@ export default class TableClipboard extends Plugin {
 			method: evt.name
 		} );
 	}
+
+	/**
+	 * Overrides default {@link module:engine/model/model~Model#insertContent `model.insertContent()`} method to handle pasting table inside
+	 * selected table fragment.
+	 *
+	 * Depending on selected table fragment:
+	 * - If a selected table fragment is smaller than paste table it will crop pasted table to match dimensions.
+	 * - If dimensions are equal it will replace selected table fragment with a pasted table contents.
+	 *
+	 * @private
+	 * @param evt
+	 * @param {module:engine/model/documentfragment~DocumentFragment|module:engine/model/item~Item} content The content to insert.
+	 * @param {module:engine/model/selection~Selectable} [selectable=model.document.selection]
+	 * The selection into which the content should be inserted. If not provided the current model document selection will be used.
+	 */
+	_onInsertContent( evt, content, selectable ) {
+		if ( selectable && !selectable.is( 'documentSelection' ) ) {
+			return;
+		}
+
+		const tableSelection = this.editor.plugins.get( TableSelection );
+		const selectedTableCells = tableSelection.getSelectedTableCells();
+
+		if ( !selectedTableCells ) {
+			return;
+		}
+
+		// We might need to crop table before inserting so reference might change.
+		let pastedTable = getTableFromContent( content );
```

Comment:
I think that it will be enough to handle the case when the pasted content consists only of a table. There are many other cases that we might want to handle in the future like if there's some empty paragraph, some other containers, etc. But that's future.

Reply:
I've changed the bahavior and created an issue for that: #6817.